From 3b81c17152db15f4e58832d1281e45fb8a7acdd3 Mon Sep 17 00:00:00 2001
From: George Mocanu <george.mocanu@nxp.com>
Date: Thu, 7 Dec 2023 16:02:42 +0200
Subject: [PATCH] plat: s32: Update IRQ and IPC memory range

Use the MSCM#1 IRQ for SCP SCMI notifications to
SRM. The GoldVIP setup is using the MSCM#0 IRQ in
the context of the multi-core setup of the Real-time
Gateway application, without any option to change
it currently.

Accommodate the SRM mailboxes for the GoldVIP
memory map - offset them by 2MB than the default.

Issue: GVIP-1193
Signed-off-by: George Mocanu <george.mocanu@nxp.com>
---
 plat/nxp/s32/include/s32_platform_def.h |  8 ++++----
 plat/nxp/s32/s32_bl31.c                 |  4 ++--
 plat/nxp/s32/s32_scp_scmi.c             | 10 +++++-----
 3 files changed, 11 insertions(+), 11 deletions(-)

diff --git a/plat/nxp/s32/include/s32_platform_def.h b/plat/nxp/s32/include/s32_platform_def.h
index c7a8cdf27..32275779d 100644
--- a/plat/nxp/s32/include/s32_platform_def.h
+++ b/plat/nxp/s32/include/s32_platform_def.h
@@ -87,8 +87,8 @@
 /* SGI to use for kicking the secondary cores out of wfi */
 #define S32_SECONDARY_WAKE_SGI	15
 /* Used for SCP notifications */
-#define S32CC_MSCM_CORE_0_IRQ	33
-#define S32CC_MAX_IRQ		(S32CC_MSCM_CORE_0_IRQ + 1)
+#define S32CC_MSCM_CORE_1_IRQ	34
+#define S32CC_MAX_IRQ		(S32CC_MSCM_CORE_1_IRQ + 1)
 
 #define S32_SRAM_BASE		0x34000000
 #define S32_SRAM_END		(S32_SRAM_BASE + S32_SRAM_SIZE)
@@ -219,10 +219,10 @@
  * ------------------
  */
 
-/* Placed at 5MB offset to avoid overlaps, as some drivers require
+/* Placed at 7MB offset to avoid overlaps, as some drivers require
  * reserved areas at the beginning of the SRAM memory.
  */
-#define S32_SCP_SCMI_MEM		(0x34500000U)
+#define S32_SCP_SCMI_MEM		(0x34700000U)
 #define S32_SCP_CH_MEM_SIZE		(128)
 #define S32_SCP_SCMI_MEM_SIZE	(S32_SCP_CH_MEM_SIZE * PLATFORM_CORE_COUNT)
 
diff --git a/plat/nxp/s32/s32_bl31.c b/plat/nxp/s32/s32_bl31.c
index 6cb131f82..8d19d1d19 100644
--- a/plat/nxp/s32/s32_bl31.c
+++ b/plat/nxp/s32/s32_bl31.c
@@ -412,7 +412,7 @@ static interrupt_prop_t *register_cpu_wake_irq(interrupt_prop_t *itr,
 static interrupt_prop_t *register_scp_notif_irq(interrupt_prop_t *itr,
 						const interrupt_prop_t *end)
 {
-	interrupt_prop_t irq_prop = INTR_PROP_DESC(S32CC_MSCM_CORE_0_IRQ,
+	interrupt_prop_t irq_prop = INTR_PROP_DESC(S32CC_MSCM_CORE_1_IRQ,
 						   GIC_HIGHEST_SEC_PRIORITY,
 						   INTR_GROUP0,
 						   GIC_INTR_CFG_EDGE);
@@ -577,7 +577,7 @@ void bl31_plat_runtime_setup(void)
 		s32cc_el3_interrupt_config();
 
 		/* Route the irq to any available core */
-		plat_ic_set_spi_routing(S32CC_MSCM_CORE_0_IRQ,
+		plat_ic_set_spi_routing(S32CC_MSCM_CORE_1_IRQ,
 					INTR_ROUTING_MODE_ANY,
 					read_mpidr());
 	} else {
diff --git a/plat/nxp/s32/s32_scp_scmi.c b/plat/nxp/s32/s32_scp_scmi.c
index cf370617f..1c5211084 100644
--- a/plat/nxp/s32/s32_scp_scmi.c
+++ b/plat/nxp/s32/s32_scp_scmi.c
@@ -19,13 +19,13 @@
 #include <plat/common/platform.h>
 #include <s32_scp_scmi.h>
 
-/* M7 Core0, Interrupt 0 will be used to send SCMI messages */
+/* M7 Core0, Interrupt 1 will be used to send SCMI messages */
 #define TX_CPN		(4u)
-#define MSCM_TX_IRQ	(0u)
+#define MSCM_TX_IRQ	(1u)
 
-/* A53 Core0, Interrupt 0 will be used to send SCMI messages */
+/* A53 Core0, Interrupt 1 will be used to send SCMI messages */
 #define RX_CPN		(0u)
-#define MSCM_RX_IRQ	(0u)
+#define MSCM_RX_IRQ	(1u)
 
 #define SCMI_GPIO_ACK_IRQ	(0xFFu)
 #define MAX_INTERNAL_MSGS	(1)
@@ -201,7 +201,7 @@ void scp_scmi_init(bool request_irq)
 	if (!request_irq)
 		return;
 
-	ret = request_intr_type_el3(S32CC_MSCM_CORE_0_IRQ,
+	ret = request_intr_type_el3(S32CC_MSCM_CORE_1_IRQ,
 				    mscm_interrupt_handler);
 	if (ret) {
 		ERROR("Failed to request MSCM interrupt\n");
-- 
2.25.1

